### PR TITLE
Autonomous tweaks

### PR DESCRIPTION
**Pull request by james-ward**
Added the flipping for alliance colour. Also changed starting position to be in front of speaker.

There is still a big bug in the generation of trajectories. If you start the sim, then select 2 note auto and run straight away, it will throw an error in the trajectory generation. But if you drive around a bit first in teleop it sometimes works, sometimes fails.

Even then, the trajectories have weird circles at the end of the path.

Help requested!!

### CODE REVIEW

**Comment by james-ward** [comment]
I figured it out - bad use of headings when making trajectories for holonomic drivebase.

File: autonomous/autonomous.py [hunk 1]
```diff
@@ -101,78 +137,69 @@ def drive_to_pick_up(self, state_tm: float, initial_call: bool) -> None:
...
+        translation = self.goal - prev_pos
+
+        spline_goal_momentum_x = translation.x * self.kD
+        spline_goal_momentum_y = translation.y * self.kD
+        goal_spline = Spline3.ControlVector(
+            (self.goal.X(), spline_goal_momentum_x),
+            (self.goal.Y(), spline_goal_momentum_y),
+        )
+
+        traj_config.setStartVelocity(0.0)
```
Comment (mlists) [inline autonomous/autonomous.py:197]: I am a little concerned about this assumption as we may exit trajectories early if we have picked up a note. Something we should keep an eye on when testing on the real robot.

**Comment by mlists** [comment]
I think we should also consider storing the note positions and our preferred firing positions (if relevant) in a way that lets us label them (perhaps a NotePositions dataclass?). It would help to reduce the number of magic coordinates and allow us to change firing positions globally.

File: autonomous/autonomous.py [hunk 2]
```diff
@@ -101,78 +137,69 @@ def drive_to_pick_up(self, state_tm: float, initial_call: bool) -> None:
...
         try:
             trajectory = TrajectoryGenerator.generateTrajectory(
                 start_point_spline, waypoints, goal_spline, traj_config
             )
-        except RuntimeError:
+        except Exception:
             return Trajectory([Trajectory.State(0, 0, 0, pose)])
```
[the remark this answers is not in the document]
Comment (james-ward) [reply autonomous/autonomous.py:203]: I cannot find anything in the documentation about how to log stuff. Print statements don't work in the sim - I tried.

[the remark this answers is not in the document]
Comment (LucienMorey) [reply autonomous/autonomous.py:201]: Does the logging module work? I was pretty sure @auscompgeek recommended using that over standard print statements anyway

[the remark this answers is not in the document]
Comment (james-ward) [reply autonomous/autonomous.py:196]: Probably. But as I said, I cannot find that in the robotpy docs. If someone has a pointer to it, I'll do it.
Also there is an error message printed anyway, this just stops the crash.

[the remark this answers is not in the document]
Comment (auscompgeek) [reply autonomous/autonomous.py:196]: Print statements should go to the console as normal, but magicbot injects a `logger: logging.Logger` into every component and auto routine. (I was fairly certain this was documented…)

**Comment by james-ward** [comment]
> I think we should also consider storing the note positions and our preferred firing positions (if relevant) in a way that lets us label them (perhaps a NotePositions dataclass?). It would help to reduce the number of magic coordinates and allow us to change firing positions globally.

That's a good idea, but beyond the scope of this PR. Perhaps do this next?

File: autonomous/autonomous.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (james-ward) [reply autonomous/autonomous.py:196]: This already prints so I am resolving this comment

**Review by mlists — APPROVED** [review]
I couldn't see the exceptions I was raising myself in the try-except and I am not sure how to cause the genuine error state but otherwise looks good.

**Pull request merged**